app/assets/javascripts/utils/api.js

```diff
@@ -600,6 +598,28 @@ const API = {
         label,
       };
     });
+  },
+
+  async checkArticleInWikiCategory(wiki_articles_title) {
+    const category_members = [];
+    for (const title of wiki_articles_title) {
+      try {
+        const response = await request(`/assignments/discouraged/${title}`);
+
+        if (response.ok) {
+          const { is_category_member } = await response.json();
+          if (is_category_member) {
+            category_members.push(title);
+          }
+        } else {
+          throw new Error('Request failed');
+        }
+      } catch (error) {
+        logErrorMessage(error);
+      }
+    }
+
+    return category_members;
```

Comment:
is there any reason this can't be parallelised? Currently, only request is made at a time even though all the requests are independent of each other.

Reply:
It can be parallelised 😅. Additionally, I am considering sending a single POST request to the server, passing the entire `wiki_articles_title` array in the request body?

Reply:
that sounds better